src/character_attire.cpp

```diff
@@ -1945,6 +1950,171 @@ void outfit::absorb_damage( Character &guy, damage_unit &elem, bodypart_id bp,
+                        int fire_intensity = std::ceil( intensity * ( liquid_remaining / liquid_amount ) );
+                        if( fire_intensity >= 1 ) {
+                            fire_data frd{ fire_intensity };
+                            destroy = !armor.has_flag( flag_INTEGRATED ) && armor.burn( frd );
+                            int fuel = roll_remainder( frd.fuel_produced );
+                            if( fuel > 0 ) {
+                                guy.add_effect( effect_onfire, time_duration::from_turns( fuel + 1 ), bp, false, 0, false,
+                                                true );
+                            }
+                        }
+                    }
+                    if( !destroy ) {
+                        // The roll here is 0 because we already rolled to see if the attack hurts the armor.
+                        // Not -1 so that we can't splash items with 0 coverage.
+                        if( secondary_sbp != sub_bodypart_id() ) {
+                            // Check for items hanging from the bodypart we hit.
+                            destroy = guy.armor_absorb( damage, armor, bp, secondary_sbp, 0 );
+                        } else {
+                            destroy = guy.armor_absorb( damage, armor, bp, sbp, 0 );
+                        }
+                    }
+                    if( destroy ) {
+                        if( get_player_view().sees( guy ) ) {
+                            SCT.add( point( guy.posx(), guy.posy() ), direction::NORTH, remove_color_tags( pre_damage_name ),
+                                     m_neutral, _( "destroyed" ), m_info );
+                        }
+                        destroyed_armor_msg( guy, pre_damage_name );
+                        armor.on_takeoff( guy );
+                        for( const item *it : armor.all_items_top( pocket_type::CONTAINER ) ) {
+                            worn_remains.push_back( *it );
+                        }
+                        iter = decltype( iter )( worn.erase( --iter.base() ) );
+                    }
+                }
+            }
+            // Whether or not the item was affected by the fluid, it still blocked some or all of it.
+            // Breathability and coverage let fluid soak through, but some is lost, weakening the attack as it goes.
+            liquid_remaining = std::max( 0,
+                                         liquid_remaining - ( ( armor.get_coverage( bp ) + armor.breathability( bp ) ) / 2 ) );
+            damage.amount *= static_cast<float>( liquid_remaining ) / static_cast<float>( liquid_amount );
+        }
+        ++iter;
+    }
+    if( spell_effect != effect_null ) {
+        intensity = std::ceil( intensity * ( liquid_remaining / liquid_amount ) );
+        if( intensity >= 1 ) {
+            guy.add_effect( spell_effect, dur_td, bp, permanent, intensity );
+        }
+    }
+    if( liquid_remaining == liquid_amount && !guy.is_avatar() ) {
```

Comment:
is _avatar() check is redundant (and erroneous), this message can be handled with Character::add_msg_player_or_npc. Assuming that you are working around the vision check, it would still be better to opt it with Character::add_msg_if_player